src/transformers/tokenization_utils.py

```diff
@@ -572,7 +572,7 @@ def _add_tokens(self, new_tokens: Union[List[str], List[AddedToken]], special_to
             self._added_tokens_encoder[token.content] = token_index
             if self.verbose:
                 logger.info(f"Adding {token} to the vocabulary")
-
+            self.total_vocab_size += 1
```

Comment:
we should only add if `token.content not in current_vocab` !

Reply:
Also probably in the `    @added_tokens_decoder.setter
    def added_tokens_decoder(self, value: Dict[int, Union[AddedToken, str]]) -> Dict[int, AddedToken]:`

Reply:
as we don't add the tokens but set them!

Reply:
I reverted to an earlier commit that addresses these better! @ArthurZucker